tensorflow_recommenders_addons/dynamic_embedding/python/ops/dynamic_embedding_creator.py

```diff
@@ -225,3 +243,150 @@ def get_config(self):
         'device': self.device,
     }
     return config
+
+
+# SaveableObjectCreator
+class SaveableObjectCreator(object, metaclass=ABCMeta):
+  """
+      
+
+    Example usage:
+
+    ```python
+    saveable_config=
+    ```
+  """
+  _upsert_restore = True
+
+  def __init__(self, config=None):
+    self.config = config
+
+  def create_variable_saveable_object(self, variable, name):
+    raise NotImplementedError('create function must be implemented')
+
+  def create_shard_saveable_object(self,
+                                   variable,
+                                   table,
+                                   shard_idx,
+                                   name,
+                                   full_name=""):
+    raise NotImplementedError('create function must be implemented')
+
+  def set_upsert_restore(self, setting):
+    self._upsert_restore = setting
+
+
+class FileSystemSaveableObjectConfig(object):
+
+  def __init__(self,
+               mpi_size: int = None,
```

Comment:
`table_id` or `table_size`

Reply:
proc_rank and proc_size following TF style?